portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,20 +29,35 @@
 
 /** Servlet that allows users to add comments and returns comment history  */
 @WebServlet("/comments")
-public class DataServlet extends HttpServlet {
-    
+public class DataServlet extends HttpServlet {  
+  
+  private static int maxComments;
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
+    String maxCommentsString = request.getParameter("numcomments");
+
+    // Convert the input to an int.
+    try {
+        maxComments = Integer.parseInt(maxCommentsString);
+    } catch (NumberFormatException e) {
+        System.err.println("Could not convert to int: " +  maxCommentsString);
+    }
+
     ArrayList<String> comments = new ArrayList<>();
 
+    int currentCount = 0;
+
     for (Entity entity : results.asIterable()) {
+      if (currentCount >= maxComments) break;
```

Comment:
Can we make Datastore do the work of limiting the number of comments returned, instead of tracking it ourselves here? It's better to get only the data we need from the database.

Reply:
Sounds good! I'll read the documentation to see how to achieve this and change it.